transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringCompletionQueue.java

```diff
@@ -90,9 +90,9 @@ int process(IOUringCompletionQueueCallback callback) {
             PlatformDependent.putIntOrdered(kHeadAddress, ringHead);
 
             int fd = (int) (udata >>> 32);
-            int opMask = (int) (udata & 0xFFFFFFFFL);
-            int op = opMask >>> 16;
-            int data = opMask & 0xffff;
+            int opMask = (int) udata;
+            int op = (short) (opMask >> 16);
+            int data = (short) opMask;
```

Comment:
I can also do this as an extra PR if you prefer... @njhill @chrisvest

Reply:
Maybe you could you explain this one.. I may just not understand but thought here we basically want to slice up 64 bits in an unsigned manner? I'm not sure casting long to int / int to short is the best way to do that... in particular if the MSB is 1 then it will get mangled. It already looked right to me before (apart from the `f`/`F` inconsistency!).

Reply:
the problem was that we did not handle negative values correctly. Like if we did use `-1` as `data`. Let me do a separate PR with a test case to make it more clear

Reply:
@njhill https://github.com/netty/netty/pull/10617